Add support for GoDaddy DNS, update to Alpine 3.14 and all custom authentication components

irakhlin opened the pull request:
- Update image to Alpine 3.14
- Update certbot to 1.19
- Update DirectAdmin DNS to 0.0.23
- Update Netcup DNS to 1.1.1
- Update Njalla DNS to 1.0.0
- Update Gandi DNS to 1.3.2
- Update Transip DNS to 0.3.3

XantXant commented:
You have an other solution for cryptography issue.
I added INWX dns provider, but i changed the apk list to install.
So cryptography can build with rust.
But now certbot-dns-transip is not working in my solution, because of the problem with suds-jurko.

[https://github.com/XantXant/addons](https://github.com/XantXant/addons)

I will wait with my pull request.

irakhlin commented:
> You have an other solution for cryptography issue.
> 
> I added INWX dns provider, but i changed the apk list to install.
> 
> So cryptography can build with rust.
> 
> But now certbot-dns-transip is not working in my solution, because of the problem with suds-jurko.
> 
> 
> 
> [https://github.com/XantXant/addons](https://github.com/XantXant/addons)
> 
> 
> 
> I will wait with my pull request.

I handled the issue with suds-jurko, and cryptography by leveraging the already compiled cryptography wheel provided by home assistant. If there is any desire to have this merged in I would be happy to look at adding INWX

pvizeli commented on letsencrypt/Dockerfile:
```diff
@@ -18,8 +19,20 @@ RUN apk add --no-cache --update \
+        esac \
     && pip3 install --no-cache-dir --find-links \
-        "https://wheels.home-assistant.io/alpine-$(cut -d '.' -f 1-2 < /etc/alpine-release)/${BUILD_ARCH}/" \
+        "https://wheels.home-assistant.io/alpine-$(cut -d '.' -f 1-2 < /etc/alpine-release)/${ARCH}/" \
```
Why you change that?

irakhlin replied:
Watching the build I saw zero prebuilt wheels being pulled from the home assistant wheel repository. When looking into it further it seemed the arch type was always resolving to "empty".

pvizeli replied:
Could be because the ARCH arg is missing above

irakhlin replied:
Do you mean the BUILD_ARCH? I was not able to get it properly working, maybe it was something to do with my pipeline? Regardless I have tested the current implementation and it works fine for all the architectures.

pvizeli replied:
Right, but we use BUILD_ARCH and the issue is, it's just not added on the file there. You can't use an arg if that is not specified before. That is the issue, I don't say you did the issue

irakhlin replied:
My misunderstanding, sorry!
If I am getting you correctly I should be able to add:
ARG BUILD_ARCH 
then I can revert the URL to use ${BUILD_ARCH} and get rid of this entire section
`apkArch="$(apk --print-arch)"; \
       case "$apkArch" in \
         armhf) export ARCH='armhf' ;; \
         x86) export ARCH='i386' ;; \
         armv7) export ARCH='armv7' ;; \
         x86_64) export ARCH='amd64' ;; \
         aarch64) export ARCH='aarch64' ;; \
        esac \`

I am guessing that gets passed as part of the CI build process?

pvizeli replied:
right

irakhlin replied:
I'll try to push that up later tonight.

irakhlin replied:
pushed up the changes we discussed

pvizeli commented on letsencrypt/Dockerfile:
```diff
@@ -18,8 +19,20 @@ RUN apk add --no-cache --update \
+    && pip3 install setuptools==57.5.0 \
+    && echo "cffi==1.14.6" >> /tmp/requirements.txt \
+    && echo "cryptography==3.4.8" >> /tmp/requirements.txt \
+    && echo "setuptools==57.5.0" >> /tmp/requirements.txt \
```
Why is that explizit needed?

irakhlin replied:
setuptools passed the specified version no longer allows 2to3 as part of the wheel build command. This breaks the package that Transip depends on suds-jurko. 57.5.0 is the latest version without that "feature" https://setuptools.pypa.io/en/latest/history.html#v58-0-0

As far as pinning the version of cryptography along with the fix to correctly getting the ARCH type allows me to pull the prebuilt newest wheel available from the home assistant wheel repo. Without this we would need to introduce the Rust installer script as the version of rust on alpine is currently too old to build cryptography. 

Let me know if you have any follow up questions.

pvizeli replied:
Could be because the ARCH arg is missing above

pvizeli replied:
You can just get the cryptography library from our wheel repo

irakhlin replied:
that is exactly where it's coming from. Without pinning it to a version however one of the many dependencies can try and request a "newer" one and the one from the repo gets ignored. the intent here is to make sure we grab it from the repo.

homeassistant commented:
Hello @irakhlin,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`11f87708f42ef521142e1f0a2de17d595cead1e8`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;addons&#x2F;pull&#x2F;2177&#x2F;commits&#x2F;11f87708f42ef521142e1f0a2de17d595cead1e8) - This commit has something that looks like an email address (irakhlin@ptc.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

irakhlin commented:
> 
> 
> Hello @irakhlin,
> 
> When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).
> 
> The commits that are missing a linked GitHub account are the following:
> 
>     * [`11f87708f42ef521142e1f0a2de17d595cead1e8`](https://github.com/home-assistant/addons/pull/2177/commits/11f87708f42ef521142e1f0a2de17d595cead1e8) - This commit has something that looks like an email address ([irakhlin@ptc.com](mailto:irakhlin@ptc.com)). Maybe try linking that to GitHub?.
> 
> 
> Unfortunately, **we are unable to accept this pull request until this situation is corrected.**
> 
> Here are your options:
> 
>     1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).
> 
>     2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
>        
>        * If you only made a single commit you should be able to run
>          ```
>          git commit --amend --author="Author Name <email@address.com>"
>          ```
>          
>          
>              
>                
>              
>          
>                
>              
>          
>              
>            
>          (substituting Author Name and [email@address.com](mailto:email@address.com) for your actual information) to set the authorship information.
>        * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
>          
>          1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
>          2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
>        * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.
> 
> 
> We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.
> 
> Thanks, I look forward to checking this PR again soon! ❤️

This was my mistake! I went ahead and amended my previous commit with my correct github email and name. I believe it should be all set, I made sure the 2nd commit had all the needed information. Please let me know if its not correct.

ludeeus reviewed (changes requested):
This PR contains unrelated commits.
Please clean, or reopen a new PR (Generally it's not a good idea to use "master" from your fork when creating PR's)

irakhlin commented:
closing and recreating
